### ISSUE TITLE
eslint-disable-line directive still errors for exceeding max-len

### ISSUE THREAD

**Issue by kentor** [eslint/eslint#2074]
```javascript
1; // eslint-disable-line
```

```yaml
---
rules:
  max-len: [2, 1]
```

```bash
index.js
  1:0  error  Line 1 exceeds the maximum line length of 1  max-len

✖ 1 problem (1 error, 0 warnings)
```

Expected the directive to allow exceeding max-len rule.

**Comment by nzakas** [comment]
Ah so, this actually looks like it's behaving correctly. The maximum line length includes the comment.

**Comment by hon2a** [comment]
@nzakas Maybe I don't quite get what you mean, but how do you figure that? When I do:

```
... some code longer than allowed max-len ...;  // eslint-disable-line max-len
```

I'd expect the comment to disable linting of that line and no longer return the warning/error for that line. However, when I try it, I'm still getting my warning. If I disable the rule like this:

```
/* eslint-disable max-len */
... some code longer than allowed max-len ...;
/* eslint-enable max-len */
```

it works as expected (no warning).

**Comment by oluckyman** [comment]
same here.
This code throws an error
```
... some code longer than allowed max-len ...;  // eslint-disable-line max-len
```
this code doesn’t throw an error
```
/* eslint-disable max-len */
... some code longer than allowed max-len ...;
```
seems like a bug

**Comment by mkscrg** [comment]
Just hit this myself. Definitely seems like a bug. @nzakas maybe this was closed on a misunderstanding?

**Comment by ilyavolodin** [comment]
@mkscrg There's another issue opened for it somewhere (can't find it right now). We need to do this globally for all rules and not do one-of fixes.

**Comment by zachlysobey** [comment]
I'm still seeing this issue, and not sure how to resolve.
Any update? Or link to another relevant issue?
If this is the same bug, a year later, thats unfortunate :(

**Comment by not-an-aardvark** [comment]
@zachlysobey May you please create a new issue and fill out the [bug report template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md)? If this bug still exists, it will probably get more attention in a new issue.

**Comment by gkatsanos** [comment]
same problem.

**Comment by zachlysobey** [comment]
@not-an-aardvark @ilyavolodin

I can make a new issue, but it will essentially be just a copy of this one... Maybe just better to re-open?

**Comment by davidmaxwaterman** [comment]
was this actually moved to another issue, or was it just forgotten?

**Comment by julmot** [comment]
Same issue

**Comment by MonsieurWave** [comment]
same issue

**Comment by not-an-aardvark** [comment]
May you please create a new issue and fill out the [bug report template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md)? If this bug still exists, it will probably get more attention in a new issue.

**Comment by gkatsanos** [comment]
why?

**Comment by not-an-aardvark** [comment]
I can't reproduce the bug based on the example given in the top of the issue. It's possible that there is a bug somewhere, but it's not the same as the original problem that was reported in this issue. We can't help with your issue because no one has provided enough information to reproduce the problem. If you'd like to see the bug that you're encountering get fixed, please create a new issue and fill out the [bug report template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md).
